### ISSUE TITLE
Update from 1.0.1 to 1.0.2 fails 

### ISSUE THREAD

**Issue by achurak** [revenz/FileFlows#317]
Hello, tried updating the server to 1.0.2 and it failed with db migration errors:

```
Info -> ==================================================
2022-09-12 12:21:33.449 [INFO] -> ==================================================
Info -> Database initialized
2022-09-12 12:21:33.526 [INFO] -> Database initialized
Info -> Upgrade running, running 1.0.2 upgrade script
2022-09-12 12:21:33.812 [INFO] -> Upgrade running, running 1.0.2 upgrade script
Info -> Migrating Library Files Data
2022-09-12 12:21:33.815 [INFO] -> Migrating Library Files Data
Info -> About to migrate 5906 files
2022-09-12 12:21:41.749 [INFO] -> About to migrate 5906 files
Error -> Error: constraint failed
UNIQUE constraint failed: LibraryFile.Uid
   at System.Data.SQLite.SQLite3.Reset(SQLiteStatement stmt)
   at System.Data.SQLite.SQLite3.Step(SQLiteStatement stmt)
   at System.Data.SQLite.SQLiteDataReader.NextResult()
   at System.Data.SQLite.SQLiteDataReader..ctor(SQLiteCommand cmd, CommandBehavior behave)
   at System.Data.SQLite.SQLiteCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteScalar(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteScalar()
   at NPoco.Database.<>c__DisplayClass297_0.<ExecuteScalarHelper>b__0()
   at NPoco.Database.ExecutionHook[T](Func`1 action)
   at NPoco.Database.ExecuteScalarHelper(DbCommand cmd)
   at NPoco.DatabaseTypes.SQLiteDatabaseType.ExecuteInsert[T](Database db, DbCommand cmd, String primaryKeyName, Boolean useOutputClause, T poco, Object[] args)
   at NPoco.Database.InsertAsyncImp[T](PocoData pocoData, String tableName, String primaryKeyName, Boolean autoIncrement, T poco, Boolean sync)
   at NPoco.AsyncHelper.RunSync[T](Task`1 task)
   at NPoco.Database.Insert[T](String tableName, String primaryKeyName, Boolean autoIncrement, T poco)
   at NPoco.Database.Insert[T](T poco)
   at FileFlows.Server.Upgrade.Upgrade_1_0_2.MigrateLibraryFilesData() in /src/Server/Upgrade/Upgrade_1_0_2.cs:line 41
   at FileFlows.Server.Upgrade.Upgrade_1_0_2.Run(Settings settings) in /src/Server/Upgrade/Upgrade_1_0_2.cs:line 19
   at FileFlows.Server.Upgrade.Upgrader.Run(Settings settings) in /src/Server/Upgrade/_Upgrader.cs:line 58
   at FileFlows.Server.Program.PrepareDatabase() in /src/Server/Program.cs:line 247
   at FileFlows.Server.Program.Main(String[] args) in /src/Server/Program.cs:line 107
2022-09-12 12:21:41.813 [ERRR] -> Error: constraint failed
UNIQUE constraint failed: LibraryFile.Uid
   at System.Data.SQLite.SQLite3.Reset(SQLiteStatement stmt)
   at System.Data.SQLite.SQLite3.Step(SQLiteStatement stmt)
   at System.Data.SQLite.SQLiteDataReader.NextResult()
   at System.Data.SQLite.SQLiteDataReader..ctor(SQLiteCommand cmd, CommandBehavior behave)
   at System.Data.SQLite.SQLiteCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteScalar(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteScalar()
   at NPoco.Database.<>c__DisplayClass297_0.<ExecuteScalarHelper>b__0()
   at NPoco.Database.ExecutionHook[T](Func`1 action)
   at NPoco.Database.ExecuteScalarHelper(DbCommand cmd)
   at NPoco.DatabaseTypes.SQLiteDatabaseType.ExecuteInsert[T](Database db, DbCommand cmd, String primaryKeyName, Boolean useOutputClause, T poco, Object[] args)
   at NPoco.Database.InsertAsyncImp[T](PocoData pocoData, String tableName, String primaryKeyName, Boolean autoIncrement, T poco, Boolean sync)
   at NPoco.AsyncHelper.RunSync[T](Task`1 task)
   at NPoco.Database.Insert[T](String tableName, String primaryKeyName, Boolean autoIncrement, T poco)
   at NPoco.Database.Insert[T](T poco)
   at FileFlows.Server.Upgrade.Upgrade_1_0_2.MigrateLibraryFilesData() in /src/Server/Upgrade/Upgrade_1_0_2.cs:line 41
   at FileFlows.Server.Upgrade.Upgrade_1_0_2.Run(Settings settings) in /src/Server/Upgrade/Upgrade_1_0_2.cs:line 19
   at FileFlows.Server.Upgrade.Upgrader.Run(Settings settings) in /src/Server/Upgrade/_Upgrader.cs:line 58
   at FileFlows.Server.Program.PrepareDatabase() in /src/Server/Program.cs:line 247
   at FileFlows.Server.Program.Main(String[] args) in /src/Server/Program.cs:line 107
Error: constraint failed
UNIQUE constraint failed: LibraryFile.Uid
   at System.Data.SQLite.SQLite3.Reset(SQLiteStatement stmt)
   at System.Data.SQLite.SQLite3.Step(SQLiteStatement stmt)
   at System.Data.SQLite.SQLiteDataReader.NextResult()
   at System.Data.SQLite.SQLiteDataReader..ctor(SQLiteCommand cmd, CommandBehavior behave)
   at System.Data.SQLite.SQLiteCommand.ExecuteReader(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteScalar(CommandBehavior behavior)
   at System.Data.SQLite.SQLiteCommand.ExecuteScalar()
   at NPoco.Database.<>c__DisplayClass297_0.<ExecuteScalarHelper>b__0()
   at NPoco.Database.ExecutionHook[T](Func`1 action)
   at NPoco.Database.ExecuteScalarHelper(DbCommand cmd)
   at NPoco.DatabaseTypes.SQLiteDatabaseType.ExecuteInsert[T](Database db, DbCommand cmd, String primaryKeyName, Boolean useOutputClause, T poco, Object[] args)
   at NPoco.Database.InsertAsyncImp[T](PocoData pocoData, String tableName, String primaryKeyName, Boolean autoIncrement, T poco, Boolean sync)
   at NPoco.AsyncHelper.RunSync[T](Task`1 task)
   at NPoco.Database.Insert[T](String tableName, String primaryKeyName, Boolean autoIncrement, T poco)
   at NPoco.Database.Insert[T](T poco)
   at FileFlows.Server.Upgrade.Upgrade_1_0_2.MigrateLibraryFilesData() in /src/Server/Upgrade/Upgrade_1_0_2.cs:line 41
   at FileFlows.Server.Upgrade.Upgrade_1_0_2.Run(Settings settings) in /src/Server/Upgrade/Upgrade_1_0_2.cs:line 19
   at FileFlows.Server.Upgrade.Upgrader.Run(Settings settings) in /src/Server/Upgrade/_Upgrader.cs:line 58
   at FileFlows.Server.Program.PrepareDatabase() in /src/Server/Program.cs:line 247
   at FileFlows.Server.Program.Main(String[] args) in /src/Server/Program.cs:line 107
```

Please let me know if you need any additional information. Thanks!

**Comment by revenz** [comment]
can you supply me with your sqlite database you are trying to upgrade?   via a PM on discord might be the easiest way to get it to me.

**Comment by achurak** [comment]
Thanks! I will try to grab that file and provide it to you (it's on a distributed block storage).

**Comment by revenz** [comment]
sorry didnt realise it was  you on discord asking about the DB (I had forgotten about this issue).  just include the URL for this ticket and message me again.

I should just need the FileFlows.sqlite file, or the FileFlows_1.0.1.sqlite if that exists

**Comment by achurak** [comment]
I fixed it. Apparently this db migration, 1.0.1 to 1.0.2, was taking so long the first time I tried upgrading that my k8s decided it was unhealthy, killed it and recreated. 

The new container tried running migration again but since some of the files have already been migrated by the killed container the unique constraint immediately failed.

I bumped the startup probe timer up, deleted the `FileFlows.sqlite` that was *half-migrated*, copied `FileFlows-1.0.1.sqlite.backup` to `FileFlows.sqlite` and tried upgrading again and this time it worked fine.
